### PR TITLE
[WIP] Minor performance improvement

### PR DESCRIPTION
**Pull request by Ailrun**
At least memory usage of ghcide looks better in my machine.

### CODE REVIEW

File: ghcide/src/Development/IDE/Core/Rules.hs [hunk 1]
```diff
@@ -35,7 +35,7 @@ import           Control.Concurrent.Extra
 import           Control.Exception
 import           Control.Monad.Extra
 import           Control.Monad.Reader
-import           Control.Monad.State
+import           Control.Monad.State.Strict
```
Comment (Ailrun) [inline ghcide/src/Development/IDE/Core/Rules.hs:38]: This is the only change, TBH.

Comment (pepeiborra) [reply ghcide/src/Development/IDE/Core/Rules.hs:38]: If this is the only change please revert the other white space changes please

**Comment by pepeiborra** [comment]
Did you run cabal bench ghcide and check?

File: ghcide/src/Development/IDE/Core/Rules.hs (hunk 1 above)
Comment (Ailrun) [reply ghcide/src/Development/IDE/Core/Rules.hs:38]: Is there a reason for those aligned imports and unaligned imports?

**Comment by Ailrun** [comment]
@pepeiborra I have run it, but it looks like it does not give any memory related info, does it?

**Comment by pepeiborra** [comment]
Yes, it does. Follow the links in ghcide/bench/readme or just look at the svg files in the Bench workflow artifacts

**Comment by Ailrun** [comment]
Oh, I need to check artifacts manually, thank you for letting me know.

**Comment by Ailrun** [comment]
That's weird, benchmark svgs show the same memory footprint. I will close this PR for now.

**Pull request closed without merging**